cime_config/buildnml

```diff
@@ -297,7 +304,28 @@ def buildnml(case, caseroot, compname):
             # End if
 
         # Determine location and name of "user_nl_cam" files:
-        user_nl_file = os.path.join(caseroot, "user_nl_cam" + inst_string)
+        user_nl_fname = "user_nl_cam" + inst_string
+        user_nl_file = os.path.join(caseroot, user_nl_fname)
+
+        # Temporary user_nl file with history config stripped
+        user_nl_temp = os.path.join(confdir, "user_nl_temp")
+        if os.path.exists(user_nl_temp):
+            os.remove(user_nl_temp)
+        # end if
+
+        # Remove history configuration from the normal user_nl content
+        with open(user_nl_file, 'r') as infile:
+            clines = infile.readlines()
+        # end with
+        with open(user_nl_temp, 'w') as outfile:
+            for line in clines:
+                sline = line.strip()
+                if ((not sline) or (sline[0] == '!') or
```

Comment:
```suggestion
                if (sline.startswith('!') or
```
Removes the need for a line length check.

Reply:
Actually, I just realized what this is actually doing.  Disregard!
If anything, using `not sline or sline.startswith('!')` might be a good idea but it also looks like this isn't needed anymore.  😄